### PR TITLE
KTOR-6001 Cache returns null when vary header has more fields in the cached response

### PR DESCRIPTION
**Pull request by rsinukov**
https://youtrack.jetbrains.com/issue/KTOR-6001

### CODE REVIEW

File: ktor-client/ktor-client-core/common/src/io/ktor/client/plugins/cache/storage/UnlimitedCacheStorage.kt [hunk 1]
```diff
@@ -42,7 +42,9 @@ internal class UnlimitedStorage : CacheStorage {
 
     override suspend fun find(url: Url, varyKeys: Map<String, String>): CachedResponseData? {
         val data = store.computeIfAbsent(url) { ConcurrentSet() }
-        return data.find { it.varyKeys == varyKeys }
+        return data.find {
```
Comment (marychatte) [inline ktor-client/ktor-client-core/common/src/io/ktor/client/plugins/cache/storage/UnlimitedCacheStorage.kt:45]: There is one more `find` function on line 23, should we also change it like here?

File: ktor-client/ktor-client-core/common/src/io/ktor/client/plugins/cache/storage/UnlimitedCacheStorage.kt [hunk 2]
```diff
@@ -42,7 +42,9 @@ internal class UnlimitedStorage : CacheStorage {
...
+            varyKeys.all { (key, value) -> it.varyKeys[key] == value }
```
Comment (marychatte) [inline ktor-client/ktor-client-core/common/src/io/ktor/client/plugins/cache/storage/UnlimitedCacheStorage.kt:46]: Shouldn't we also check the equal size of maps?

**Review by marychatte — COMMENTED** [review]
Sorry, accidentally missed this pr :) I have some comments

File: ktor-client/ktor-client-core/common/src/io/ktor/client/plugins/cache/storage/UnlimitedCacheStorage.kt (hunk 2 above)
Comment (rsinukov) [reply ktor-client/ktor-client-core/common/src/io/ktor/client/plugins/cache/storage/UnlimitedCacheStorage.kt:46]: No, it's ok if cached version has more vary properties, than the 304 response

File: ktor-client/ktor-client-core/common/src/io/ktor/client/plugins/cache/storage/UnlimitedCacheStorage.kt (hunk 1 above)
Comment (rsinukov) [reply ktor-client/ktor-client-core/common/src/io/ktor/client/plugins/cache/storage/UnlimitedCacheStorage.kt:47]: This is the legacy storage, but I changed it as well. Thanks!

**Review by marychatte — APPROVED** [review]
lgtm

**Pull request merged**